allennlp/common/testing/distributed_test_context_manager.py

```diff
@@ -0,0 +1,80 @@
+from typing import List
+import torch
+import torch.distributed as dist
+import torch.multiprocessing as mp
+
+from allennlp.common.checks import check_for_gpu
+
+
+class DistributedTestContextManager:
+    """
+    This context manager is for simulating a distributed environment.
+
+    # Parameters
+
+    device_ids: `List[int]`
+        List of cuda devices. There need to be at least 2 devices. Default is [-1, -1].
+
+    # Usage
+
+    ```
+    python
+    with DistributedTestContextManager(devices_id) as test_this_function:
+        test_this_function(your_distributed_func, name_of_variable_to_check, value_to_check_against)
```

Comment:
Initially there was a need to clean up stray processes. This is no longer required. That said, having a class instead of different functions seems neater.